src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -2159,6 +2159,100 @@ module Queries = struct
             return
             @@ Error "Could not obtain best chain from transition frontier" )
 
+  let account_actions =
+    field "accountActions"
+      ~doc:
+        "Find all the actions associated to an account from the current best \
+         tip."
+      ~typ:(non_null @@ list @@ non_null Types.Action_state.spec)
+      ~args:
+        Arg.
+          [ arg "publicKey" ~doc:"Public key of account being retrieved"
+              ~typ:(non_null Types.Input.PublicKey.arg_typ)
+          ; arg' "token"
+              ~doc:"Token of account being retrieved (defaults to MINA)"
+              ~typ:Types.Input.TokenId.arg_typ ~default:Token_id.default
+          ; arg "maxLength"
+              ~doc:
+                "The maximum number of blocks to search for actions. If there \
+                 are more blocks in the transition frontier from root to tip, \
+                 the maxLength blocks closest to the best tip will be returned"
+              ~typ:int
+          ]
+      ~resolve:(fun { ctx = mina; _ } () pk token max_length ->
+        let best_chain = Mina_lib.best_chain ?max_length mina in
+        match best_chain with
+        | Some best_chain ->
+            let actions =
+              List.concat_map
+                ~f:(fun bc ->
+                  let user_cmds =
+                    bc |> Transition_frontier.Breadcrumb.block
+                    |> Mina_block.body
+                    |> Staged_ledger_diff.Body.staged_ledger_diff
+                    |> Staged_ledger_diff.commands
+                  in
+                  let block_number =
+                    bc |> Transition_frontier.Breadcrumb.block
+                    |> Mina_block.header |> Mina_block.Header.blockchain_length
+                  in
+                  let transaction_seq = ref 0 in
+                  let action_list_list =
+                    List.filter_map user_cmds ~f:(fun user_cmd ->
+                        transaction_seq := !transaction_seq + 1 ;
+                        match user_cmd.data with
+                        | Zkapp_command c -> (
```

Comment:
Just realized, I think we want to filter based on user command status. Actions from failed zkapp transactions are not applied. 
Maybe take status as input? alternatively have the status in the output and return all the actions.

Reply:
done